azurelinuxagent/common/protocol/wire.py

```diff
@@ -871,40 +888,53 @@ def get_vm_settings():
                 # clear any confusion produced by the poor choice of status code.
                 if response.status == httpclient.BAD_GATEWAY:
                     error_description = "[Internal error in HostGAPlugin] {0}".format(error_description)
+
+                if 400 <= response.status <= 499:
+                    self._vm_settings_error_reporter.report_error(error_description, _VmSettingsError.ClientError)
+                elif 500 <= response.status <= 599:
+                    self._vm_settings_error_reporter.report_error(error_description, _VmSettingsError.ServerError)
+                else:
+                    self._vm_settings_error_reporter.report_error(error_description)
+
                 raise ProtocolError("GET vmSettings [correlation ID: {0}]: {1} ({2})".format(correlation_id, response.status, error_description))
```

Comment:
Protocol error had extra error message like correlation Id and status which we are not reporting back in L#842. So can you change the error description L#890 to include extra details.

Reply:
They are added in the exception message @ L#899

Reply:
yeah, my understanding is that we don't report if it's coming from ProtocolError exception. So, we lose that information.

`# _fetch_vm_settings_goal_state() does its own detailed error reporting and raises ProtocolError; do not report those`

Reply:
got it, let me fix that

Reply:
fixed (added correlation id and etag to all messages) (the status is already added by restutil.read_response_error)